backend/dataall/modules/worksheets/services/worksheet_service.py

```diff
@@ -122,3 +129,55 @@ def run_sql_query(session, uri, worksheetUri, sqlQuery):
         )
 
         return AthenaClient.convert_query_output(cursor)
+
+    @staticmethod
+    @ResourcePolicyService.has_resource_permission(RUN_ATHENA_QUERY)
+    @ResourceThresholdRepository.check_invocation_count('nlq')
+    def run_nlq(session, username, uri, prompt, worksheetUri, db_name, table_names):
+        return 'TEST'
+        # environment = EnvironmentService.get_environment_by_uri(session, uri)
+        # worksheet = WorksheetService.get_worksheet_by_uri(session, worksheetUri)
+
+        # env_group = EnvironmentService.get_environment_group(
+        #     session, worksheet.SamlAdminGroupName, environment.environmentUri
+        # )
+
+        # glue_client = GlueClient(
+        #     account_id=environment.AwsAccountId, region=environment.region, role=env_group.environmentIAMRoleArn
+        # )
+
+        # metadata = []
+        # for table in table_names:
+        #     metadata.append(glue_client.get_table_metadata(database=db_name, table_name=table))
+
+        # response = BedrockClient().invoke_model_text_to_sql(prompt, '\n'.join(metadata))
+
+        # if response.startswith('Error:'):
+        #     raise exceptions.ModelGuardrailException(response)
+        # return response
+
+    @staticmethod
+    @ResourcePolicyService.has_resource_permission(RUN_ATHENA_QUERY)
+    @ResourceThresholdRepository.check_invocation_count('nlq')
+    def analyze_text_genai(session, username, uri, worksheetUri, prompt, datasetUri, key):
+        environment = EnvironmentService.get_environment_by_uri(session, uri)
+        worksheet = WorksheetService.get_worksheet_by_uri(session, worksheetUri)
+
+        env_group = EnvironmentService.get_environment_group(
+            session, worksheet.SamlAdminGroupName, environment.environmentUri
+        )
+
+        dataset = DatasetRepository.get_dataset_by_uri(session, datasetUri)
+
+        s3_client = S3Client(
+            account_id=environment.AwsAccountId,
+            region=environment.region,
+            role=env_group.environmentIAMRoleArn,
+        )
+
+        content = s3_client.get_content(dataset.S3BucketName, key)
+        response = BedrockClient().invoke_model_process_text(prompt, content)
+
```

Comment:
I think that the handling of the `invoke_model_process_text` error should happen inside the `invoke_model_process_text` itself

Reply:
this makes sense - good call out